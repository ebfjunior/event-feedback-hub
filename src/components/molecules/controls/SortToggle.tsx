import { Button } from '@/components/ui/button';
import type { SortOption } from '@/types/api';

export type SortToggleProps = {
  value: SortOption;
  onChange: (value: SortOption) => void;
};

export function SortToggle({ value, onChange }: SortToggleProps) {
  return (
    <div className="inline-flex items-center gap-1">
      <Button variant={value === 'newest' ? 'default' : 'outline'} size="sm" onClick={() => onChange('newest')}>
        Newest
      </Button>
      <Button variant={value === 'highest' ? 'default' : 'outline'} size="sm" onClick={() => onChange('highest')}>
        Highest Rated
      </Button>
    </div>
  );
}


