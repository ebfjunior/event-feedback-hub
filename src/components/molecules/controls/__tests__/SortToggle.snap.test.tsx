import React from 'react';
import { render } from '@testing-library/react';
import { SortToggle } from '../SortToggle';

describe('SortToggle (snapshot)', () => {
	it('renders', () => {
		const { container } = render(<SortToggle value="newest" onChange={() => {}} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


