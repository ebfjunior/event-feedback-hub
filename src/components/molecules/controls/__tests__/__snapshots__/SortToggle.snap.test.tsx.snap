// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortToggle (snapshot) > renders 1`] = `
<div
  class="inline-flex items-center gap-1"
>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-8 rounded-md px-3 text-xs"
  >
    Newest
  </button>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-8 rounded-md px-3 text-xs"
  >
    Highest Rated
  </button>
</div>
`;
